### PR TITLE
Adding flags to Vtex Proxy scripts

### PR DESCRIPTION
**Pull request by nathanluiz33**

### CODE REVIEW

File: analytics/loaders/DecoAnalyticsScript.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+): Script => {
+  const transformReq = (req: Request) => {
+    const link1 =
+      '<link rel="dns-prefetch" href="https://plausible.io/api/event" />';
+    const link2 =
+      '<link rel="preconnect" href="https://plausible.io/api/event" crossorigin="anonymous" />';
+
+    const flags = getFlagsFromCookies(req);
+
+    console.log("flags", flags, req.headers);
```
Comment (matheusgr) [inline analytics/loaders/DecoAnalyticsScript.ts:23]:
```suggestion
```

Comment (matheusgr) [inline analytics/loaders/DecoAnalyticsScript.ts:23]:
```suggestion
```

File: utils/plausible_scripts.ts [hunk 2]
```diff
@@ -16,5 +16,5 @@
...
+export const exclusionPropsAndHashScript =
```
Comment (mcandeia) [inline utils/plausible_scripts.ts:19]: Why not scripts/plausible inside analytics app?

File: analytics/loaders/DecoAnalyticsScript.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  defer?: boolean;
+};
+
+const loader = (
+  props: Props,
+  _req: Request,
+  _ctx: AppContext,
+): Script => {
+  const transformReq = (req: Request) => {
+    const link1 =
```
Comment (mcandeia) [inline analytics/loaders/DecoAnalyticsScript.ts:16]:
```suggestion
    const dnsPrefetchLink =
```

File: analytics/loaders/DecoAnalyticsScript.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      '<link rel="dns-prefetch" href="https://plausible.io/api/event" />';
+    const link2 =
```
Comment (mcandeia) [inline analytics/loaders/DecoAnalyticsScript.ts:18]:
```suggestion
    const preconnectLink =
```

**Review by mcandeia — CHANGES_REQUESTED** [review]

File: utils/cookie.ts [hunk 5]
```diff
@@ -1,4 +1,20 @@
-import { getSetCookies, setCookie } from "std/http/cookie.ts";
+import { getCookies, getSetCookies, setCookie } from "std/http/cookie.ts";
+
+export const getFlagsFromCookies = (req: Request) => {
+  const flags = [];
+  const cookies = getCookies(req.headers);
+
+  for (const [key, value] of Object.entries(cookies)) {
+    if (key.startsWith("deco_matcher_")) {
```
Comment (mcandeia) [inline utils/cookie.ts:8]: You should import from deco/block/matchers so you can guarantee that the version always is up to date with cookie name

File: utils/cookie.ts [hunk 6]
```diff
@@ -1,4 +1,20 @@
...
+      const flagName = atob(
+        value.slice(value.indexOf("=") + 1, value.indexOf("@")),
+      );
+      const flagValue = value.at(-1) === "1" ? true : false;
```
Comment (mcandeia) [inline utils/cookie.ts:12]:
```suggestion
      const flagValue = value.at(-1) === "1";
```

File: utils/cookie.ts [hunk 7]
```diff
@@ -1,4 +1,20 @@
...
+      flags.push({ flagName, flagValue });
```
Comment (mcandeia) [inline utils/cookie.ts:13]:
```suggestion
      flags.push({ name, active });
```

**Review by mcandeia — CHANGES_REQUESTED** [review]

File: utils/plausible_scripts.ts (hunk 2 above)
Comment (matheusgr) [reply utils/plausible_scripts.ts:19]: It was also used by website apps (not anymore). Moving it now.

**Review by mcandeia — APPROVED** [review]

**Pull request merged**